### PR TITLE
74703: Update UploadsController to upload a metadata.json file as wel…

### PR DESCRIPTION
**Pull request by balexandr**
## Summary

This PR updates the upload logic for IVC Forms to upload a metadata JSON file to S3 along with the PDF.

- Updated and renamed `upload_to_ivc_s3` to be more agnostic
- Added `handle_ivc_uploads` to handle multiple uploads with proper status codes
- Updated `upload_to_ivc_s3` to use numerical status codes
- Updated `resource` in `SimpleFormsApiSubmission::S3` to not need a linter skip
- Team: IVC-CHAMPVA
- Flipper: N/A

## Related Issues
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/74703

## Testing done
- Ran specs
- Manual browser testing

## Screenshots
Local S3 used for testing (In newer test `-tmp` is removed from the metadata file naming convention)
Screenshot 2024-02-20 at 2 32 33 PM

## What areas of the site does it impact?
*Since 10-10D isn't in rotation nothing else should be affected.*

### CODE REVIEW

**Review by michaelclement — APPROVED** [review]
LGTM!

**Review by reiting — APPROVED** [review]

File: lib/simple_forms_api_submission/s3.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def client
     end
 
     def resource
-      @resource ||= Aws::S3::Resource.new(client: client) # rubocop:disable Style/HashSyntax
+      @resource ||= Aws::S3::Resource.new(client:)
     end
```
Comment (balexandr) [inline lib/simple_forms_api_submission/s3.rb:38]: @cloudmagic80 From your last PR review

Comment (cloudmagic80) [reply lib/simple_forms_api_submission/s3.rb:38]: Looks okay!

**Review by cloudmagic80 — APPROVED** [review]

**Comment by cloudmagic80** [comment]
Hey Bryan. How did you set up the env file for AWS?

**Review by rjohnson2011 — APPROVED** [review]

**Review by cloudmagic80 — APPROVED** [review]

**Pull request merged**